### PR TITLE
feat(sink): upsert jdbc sink must specify downstream primary key

### LINKED ISSUE #10955: Upsert Sink: Allow users provide the downstream primary key in the WITH option
### Is your feature request related to a problem? Please describe.

Currently, when user create a upsert sink upon a mview, the primary key of downstream table may not be covered in the distribution key of mview. Thus when multiple instances of sink executor write to downstream it may cause contention for a specific primary key on downstream.

### Describe the solution you'd like

Allow users provide the downstream primary key in the WITH option when create a upsert sink, then the plan of streaming job can satisfy the downstream distribution based on this information.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Upsert JDCB sink must specify downstream primary key via `primary_key` option
- JDBC connector will validate whether the `primary_key` match the definition on downstream table

```sql
create table t1 (v1 int, v2 double precision, v3 varchar, v4 bigint, v5 decimal, primary key (v3,v4));
create sink s1_mysql as select v1, v2, v3, v5 from t1 WITH (
    connector='jdbc',
    primary_key='v1,v2',
    jdbc.url='jdbc:mysql://localhost:8306/mydb?user=root&password=xxxx',
    table.name='t1sink',
    type='upsert',
);
```

close #10955 

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Connector (sources & sinks)
- SQL commands, functions, and operators

### Release note

Upsert JDCB sink must specify the downstream primary key via `primary_key` option, which should be covered in the SELECT columns in the Sink DDL.

</details>

### CODE REVIEW

**Review by wenym1 — COMMENTED** [review]
Rest LGTM.

**Review by st1page — COMMENTED** [review]
optimizer part LGTM

**Review by fuyufjh — APPROVED** [review]
LGTM

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkConfig.java [hunk 1]
```diff
@@ -35,6 +35,9 @@ public class JDBCSinkConfig extends CommonSinkConfig {
     @JsonProperty(value = "schema.name")
     private String schemaName;
 
+    @JsonProperty(value = "primary_key")
```
[the remark this answers is not in the document]
Comment (StrikeW) [reply java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkConfig.java]: fixed

**Review by wenym1 — APPROVED** [review]

**Pull request merged**